### PR TITLE
Activate TextSelection on all interesting values/results

### LINKED ISSUE #223: The application should allow text selections on calculation results.
Ctrl+C is already supported in the app, but:
- not available on tablets/devices without keyboard.
- It's not really easy to guess that you can Ctrl+C without selecting the result.
- You can't copy an operation in the history
- You can't copy a part of a result (to ignore the fractional part of a number for example)
- If we allow users to select texts, the application will feel more like a "Desktop app", will be easier to use on a tablet and users will be more prompt to use Ctrl+C to use the result in another app.

Here is a prototype:
54065190-025b8280-41d2-11e9-8869-4029cf4532ea

### PR DESCRIPTION
**Pull request by rudyhuyn**
Fixes #223 

Ctrl+C is already supported in the app, but:
- not Tablet/Device without keyboard-friendly
- It's not really easy to guess that you can Ctrl+C without selecting the result.
- You can't copy an operation in the history
- You can't copy a part of a result (to ignore the fractional part of a number for example)

If we allow users to select texts, the application will feel more like a "Desktop app", will be easier to use on a tablet and users will be more prompt to use Ctrl+C to use the result in another app. 

2702B760-4B01-4F28-B444-05728DA30F76

## Note:
I didn't enabled TextSelection on `CalculationResult` when the control isn't "active" (text in bold), if we do,  the textblock can't catch the `Tap` events on the textblock, needed to allow users to activate the control.

### CODE REVIEW

**Comment by HowardWolosky** [comment]
Rudy, again, you're _rocking_ these changes.  Please do keep them coming.  As I've been commenting on your other PR's just now, our team is standardizing on a flow where a _behavior change_ needs to have an associated Issue which will provide the opportunity for discussion and approval before we'll consider merging in the corresponding code change.

We are in the process of updating CONTRIBUTING.md and our PR templates to clarify this.  In the meantime, please open an Issue to track this so that we can start the discussion.  Our team members will be able to join in on the discussion on Monday.

Again, _please_ keep the contributions coming.  I'm really looking forward to getting in all of these changes once we've had a bit more time to talk them through.

**Comment by HowardWolosky** [comment]
Closing this PR until it's associated with an approved issue. Locking the PR so that conversation on the problem being addressed is discussed in the Issue, so that future PR discussion can be limited to the implementation of the fix.

File: src/Calculator/Views/MemoryListItem.xaml [hunk 1]
```diff
@@ -59,6 +59,7 @@
                        Style="{ThemeResource TitleTextBlockStyle}"
                        FontWeight="SemiBold"
                        FlowDirection="LeftToRight"
+                       IsTextSelectionEnabled="True"
```
Comment (danbelcher-MSFT) [inline src/Calculator/Views/MemoryListItem.xaml]: As with the history items, recommending `HorizontalAlignment="Right"` here.

File: src/Calculator/Views/HistoryList.xaml [hunk 2]
```diff
@@ -66,13 +66,15 @@
...
                                    AutomationProperties.Name="{x:Bind AccExpression}"
+                                   IsTextSelectionEnabled="True"
                                    Text="{x:Bind Expression}"
                                    TextAlignment="Right"
                                    TextWrapping="Wrap"/>
                         <TextBlock x:Name="resultTextBlock"
                                    Style="{ThemeResource TitleTextBlockStyle}"
                                    FontWeight="SemiBold"
                                    AutomationProperties.Name="{x:Bind AccResult}"
+                                   IsTextSelectionEnabled="True"
```
Comment (danbelcher-MSFT) [inline src/Calculator/Views/HistoryList.xaml]: The `exprTextBlock` and `resultTextBlock` both stretch across the width of the ListViewItem.  After text selection is enabled, this makes it extremely difficult to click on the item itself, as the only available hit target is the thin margin between the two textblocks.
image 

We should apply `HorizontalAlignment="Right"` to both textblocks here, so as to minimize their hit target while still enabling text selection when they are explicitly hovered.

File: src/Calculator/Views/DateCalculator.xaml [hunk 3]
```diff
@@ -593,7 +595,8 @@
             <!-- Resulting Date -->
             <TextBlock x:Uid="DateLabel"
                        Grid.Row="8"
-                       Foreground="{ThemeResource SystemControlPageTextBaseMediumBrush}"/>
+                       Foreground="{ThemeResource SystemControlPageTextBaseMediumBrush}"
+                       IsTextSelectionEnabled="True"/>
```
Comment (danbelcher-MSFT) [inline src/Calculator/Views/DateCalculator.xaml]: Was this a mistake?  None of the other labels have text selection enabled so this feels out-of-place.

File: src/Calculator/Views/MemoryListItem.xaml (hunk 1 above)
Comment (rudyhuyn) [reply src/Calculator/Views/MemoryListItem.xaml]: Good idea!

File: src/Calculator/Views/DateCalculator.xaml (hunk 3 above)
Comment (rudyhuyn) [reply src/Calculator/Views/DateCalculator.xaml]: This one is not a standard label, it displays the result of the feature "Add or substract days"

Comment (danbelcher-MSFT) [reply src/Calculator/Views/DateCalculator.xaml]: I'm not sure I understand why that makes this label special?
In this first image, the red shows selectable text and the blue shows non-selectable labels:
image

Compare with this second image:
image

I don't understand why the 'Date' label is selectable text.  In the first image, 'Difference' was not selectable although it was also displaying the result of the feature.  At least we can say that we're inconsistent between each feature.  Either the label for the result should be selectable in both or not selectable in both.

Second to that, I'm arguing that the 'Date' label should not be selectable.  The *only* action the user can take by selecting the label is to copy the label alone, meaning they can't grab `Date: Wednesday, May 19, 2021` as a single run.  What is special about the label then?  What is helped by allowing it to be selectable where it wasn't before?  Why does this label deserve to be selectable but other labels like 'From' and 'To' are not selectable?

Comment (grochocki) [reply src/Calculator/Views/DateCalculator.xaml]: > I'm arguing that the 'Date' label should not be selectable. 

I agree with @danbelcher-MSFT. Field labels should not be selectable, only the values/output.

**Comment by rudyhuyn** [comment]
Oh! I was sure this comment was about the another textblock, I totally
agree, this label should not be selectable, it was a mistake

File: src/Calculator/Views/DateCalculator.xaml (hunk 3 above)
Comment (rudyhuyn) [reply src/Calculator/Views/DateCalculator.xaml]: I was confused with `DateDiffAllUnitsResultLabel`, I agree, this label should not be selectable!

Comment (danbelcher-MSFT) [reply src/Calculator/Views/DateCalculator.xaml]: 😅

**Pull request merged**